free-1.md

```diff
@@ -1,68 +1,198 @@
+something else?).
 
 ## The Project
 This section examines whether the idea makes for a good CS 111 project.
 
 
 ### Suitability
-_If someone were to work on this project, what percentage of their time would be
-spent directly engaging in the **language** aspects of this project (e.g.,
-making language design decisions), as opposed to "systems" aspects of the
-project (e.g., implementing a complicated semantics that doesn't require a lot
-of language design)?_
 
+There is some question here. I think that the time would probably end up being
+split 50/50, although that would change depending on whether I could find a
+good geometry library that would meet most of my needs (shape drawing and
+intersection detection come to mind right now, but I imagine the list will
+grow). The language design time would be spent thinking about syntax for use
+input, thinking about how the computer will provide feedback to the users, and
+thinking about the users should be able to incorporate that feedback into
+further instructions.
+
+If I were building this from the ground up this would probably be way to
+**systems** heavy.
+
+I think if on doesn't accepted the computer feedback and user response to that
+feedback as part of the language, this project would be entirely unsuitable.
+Because I think language is a 2-way street, I would say that it is suitable
+(see the "Why a Language?" section).
 
 ### Scope
-_How big an idea is this? How ambitious is this project?_
 
+It's always hard to answer this question. I think this is a project of moderate
+scope - it would not be easy, but also can be accomplished.
+
+I think that I may end up being forced into some sort of external DSL because
+of the interactive constraints, but I can probably still lean heavily on things
+like Scala for execution.
 
 ### Benefits and drawbacks
-_Why might this be a good idea for a project? Why might this not be a good idea 
-project?_
 
+I think this would be an awesome project because it would explore the idea of
+making  DSL not just a way for a user to provide instructions for the computer,
+but also a way for the computer to 'talk back'.
+
+It reminds me of a research project a friend of mine was working on, the goal
+of which was to have programs be generated after some sort of 'conversation'
+between the user and computer.
+
+I think that a drawback would be that doing all the human-computer interaction
+stuff may end up drowning out some of the finer points of syntax that DSL
+design traditionally focusses on.
+
+I also think this project is not semantically trivial. While it's not a
+deal-breaker, it would be nice to be able to focus entirely on the language
+design.
+
+I also think this project addresses a very narrow domain, and that domain may
+not feel the need for a new tool. While I would enjoy having an interactive
+system for making drawings, many users may be happy with graphical or TeX-based
+solutions.
```

Comment:
People were happy with the measuring cup until someone came up with [this](https://www.youtube.com/watch?v=8gVbo_Lz3wI)